### PR TITLE
add support for unit tests (jest + RTL)

### PR DESCRIPTION
**Pull request by SanjalKatiyar**

### CODE REVIEW

**Comment by ivoalmeida** [comment]
lgtm

**Comment by SanjalKatiyar** [comment]
/hold
need to fix some errors while running the test command

**Comment by SanjalKatiyar** [comment]
> /hold need to fix some errors while running the test command

/hold cancel

**Comment by SanjalKatiyar** [comment]
@ivoalmeida @bipuladh

File: package.json [hunk 1]
```diff
@@ -6,7 +6,9 @@
   "scripts": {
     "build": "yarn ts-node node_modules/.bin/webpack -c ./webpack.config.ts",
     "dev": "yarn ts-node node_modules/.bin/webpack serve -c ./webpack.config.ts --progress",
-    "ts-node": "ts-node -O '{\"module\":\"commonjs\"}' -I '/node_modules/(?!(@fusion)/)/'"
+    "ts-node": "ts-node -O '{\"module\":\"commonjs\"}' -I '/node_modules/(?!(@fusion)/)/'",
+    "test": "jest",
+    "test-coverage": "jest --silent --coverage --coverageReporters text-summary"
```
Comment (bipuladh) [inline package.json:11]: coverage would prolly create a folder. Do we want to add it to gitignore?

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "baseUrl": ".",
-    "module": "esnext",
```
Comment (SanjalKatiyar) [inline tsconfig.json:4]: I removed it by mistake... will add it back...

File: package.json (hunk 1 above)
Comment (SanjalKatiyar) [reply package.json:11]: missed it... will update...

File: tsconfig.json (hunk 2 above)
Comment (SanjalKatiyar) [reply tsconfig.json:4]: added "module": "esnext" back...

File: package.json (hunk 1 above)
Comment (SanjalKatiyar) [reply package.json:11]: done...

**Review by bipuladh — APPROVED** [review]

**Pull request merged**